### ISSUE TITLE
SIM114 deletes comments

### ISSUE THREAD

**Issue by k4lizen** [astral-sh/ruff#25470]
### Summary

When merging if-statements SIM114 can delete comments which should simply not happen.

See example here:
https://github.com/pwndbg/pwndbg/pull/3924/changes/c0f597c794183bd2a80e5d23057b4b5def1495b8

<img width="958" height="293" alt="Image" src="https://github.com/user-attachments/assets/0f97649e-e9b2-4f25-9de9-eee1bb7dfd0b" />

### Version

ruff 0.15.9

**Comment by dylwil3** [comment]
Minimal repro that's clickable/copyable:

```py
if cond:
    print("")
# comment
elif other:
    print("")
```

```json
{
  "preview": false,
  "lint": {
    "select": [
      "SIM114"
    ]
  },
}
```

[Playground](https://play.ruff.rs/bf22d242-7d1c-4ca6-a0fd-0af6d330d90f)

**Comment by Not-Buddy** [comment]
Can you check the PR I raised for this and let me know if its ok?
ty!
https://github.com/astral-sh/ruff/pull/25489

**Comment by ntBre** [comment]
I think this is a duplicate of https://github.com/astral-sh/ruff/issues/19576. There has been a lot of discussion on that issue and the linked PRs. This is not quite as straightforward as it seems.

**Comment by k4lizen** [comment]
oke, apologies for the noise!

**Comment by ntBre** [comment]
No worries, thanks for the report!
